src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -184,6 +184,13 @@ preflight2(PreflightContext const& ctx)
                 return temINVALID;  // LCOV_EXCL_LINE
             }
         }
+
+        if (!ctx.tx.getSigningPubKey().empty() &&
```

Comment:
Why this check is not outside of the `if (ctx.flags & tapDRY_RUN)` block? Are messing single-sign _and_ multi-sign are allowed for the real run?

Reply:
I checked that - it happens in `checkValidity` (which calls `STTx::checkSign`) for normal submission, which is skipped in this block.
https://github.com/XRPLF/rippled/blob/develop/src/libxrpl/protocol/STTx.cpp#L392-L393